File: backend/CleanArch/WebApi/Dtos/DocumentJournals.cs
using Domain.Entities;

namespace WebApi.Dtos
{
    public class CreateDocumentJournalsRequest
    {
        public string code { get; set; }
        public string name { get; set; }
        public string? number_template { get; set; }
        public int? current_number { get; set; }
        public string? reset_period { get; set; }
        public DateTime? last_reset { get; set; }
        public List<TemplateTypeOrderDto> template_types { get; set; }
        public int? period_type_id { get; set; }
        public int[] status_ids { get; set; }
    }

    public class UpdateDocumentJournalsRequest
    {
        public int id { get; set; }
        public string code { get; set; }
        public string name { get; set; }
        public string? number_template { get; set; }
        public int? current_number { get; set; }
        public string? reset_period { get; set; }
        public DateTime? last_reset { get; set; }
        public List<TemplateTypeOrderDto> template_types { get; set; }
        public int? period_type_id { get; set; }
        public int[] status_ids { get; set; }
    }
}

File: frontend/src/constants/DocumentJournals.ts
import { Dayjs } from "dayjs";

export type DocumentJournals = {
  id: number;
  code: string;
  name: string;
  number_template: string;
  current_number: number;
  reset_period: string;
  last_reset: Dayjs;
  template_types: { id: number; order: number }[];
  period_type_id: number;
  status_ids: number[];
};
